File: durable-execution/businesslogic.py
import asyncio

from temporalio import workflow


@workflow.defn
class CounterWorkflow:
    # Make the logging output look more like that of print()
    workflow.logger.workflow_info_on_message = False

    @workflow.run
    async def run(self, limit: int) -> None:
        number = 1
        while number <= limit:
            workflow.logger.warn(number)
            number = number + 1
            await asyncio.sleep(1)

File: durable-execution/businesslogic.go
package example 

import (
	"go.temporal.io/sdk/workflow"
	"fmt"
	"time"
)

func CounterWorkflow(ctx workflow.Context, limit int) error {
	logger := workflow.GetLogger(ctx)
	logger.Info("Executing Workflow")

	number := 1

	for number <= limit {
		logger.Info(fmt.Sprintf("Current Number: %d", number))
		number = number + 1
		workflow.Sleep(ctx, time.Second*1)
	}
	return nil
}
